File: src/test/programmers/java/level1/HiddenMap.java
package test.programmers.java.level1;

import java.util.Arrays;

public class HiddenMap {
    public static void main(String[] args) {
        int[] arr1 = {9, 20, 28, 18, 11};
        int[] arr2 = {30, 1, 21, 17, 28};

        Arrays.toString(solution(5, arr1, arr2));
    }

    static String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = {};
        answer = new String[n];

        for(int i = 0; i < n; i++) {
            String temp = "";
            String binary1 = Integer.toBinaryString(arr1[i]);
            while(binary1.length() < n) binary1 = "0" + binary1;

            String binary2 = Integer.toBinaryString(arr2[i]);
            while(binary2.length() < n) binary2 = "0" + binary2;

            for(int j = 0; j < n; j++) {
                temp += (binary1.charAt(j) - '0' + binary2.charAt(j) - '0' == 0) ? " " : "#";
            }

            answer[i] = temp;
        }

        return answer;
    }

}

File: src/test/programmers/python/level1/HiddenMap.py
def solution(n, arr1, arr2):
    a = []
    for i in range(n):
        s = ''
        arr1[i] = format(int(arr1[i]), 'b').zfill(n)
        arr2[i] = format(int(arr2[i]), 'b').zfill(n)

        for j in range(len(arr1)):
            if arr1[i][j] == '1' or arr2[i][j] == '1':
                s += '#'
            else:
                s += ' '
        a.append(s)

    return a

# def solution2(n, arr1, arr2):
#     answer = []
#     for i,j in zip(arr1,arr2):
#         a12 = str(bin(i|j)[2:])
#         a12=a12.rjust(n,'0')
#         a12=a12.replace('1','#')
#         a12=a12.replace('0',' ')
#         answer.append(a12)
#     return answer
